src/index.js

```diff
@@ -1 +1,128 @@
 'use strict';
+
+require('dotenv/config');
+
+const express = require('express');
+const { Server } = require('socket.io');
+const cors = require('cors');
+const MessageService = require('./services/Message.service.js');
+const UserService = require('./services/User.service.js');
+const RoomService = require('./services/Room.service.js');
+const UserController = require('./controllers/User.controller.js');
+const RoomController = require('./controllers/Room.controller.js');
+const MessageController = require('./controllers/Message.controller.js');
+const vars = require('./vars.js');
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+
+app.get('/users', UserController.getAll);
+app.get('/rooms', RoomController.getAll);
+app.get('/messages/:id', MessageController.getAllByRoomId);
+
+const server = app.listen(vars.PORT);
+const io = new Server(server, {
+  connectionStateRecovery: {
+    // the backup duration of the sessions and the packets
+    maxDisconnectionDuration: 2 * 60 * 1000,
+  },
+  cors: {
+    origin:
+      process.env.NODE_ENV === 'production'
+        ? false
+        : ['http://localhost:5173', 'http://127.0.0.1:5173'],
+  },
+});
+
+io.on('connection', (socket) => {
+  if (socket.recovered) {
+    // eslint-disable-next-line no-console
+    console.log(`User ${socket.id} reconnected`);
+  } else {
+    // eslint-disable-next-line no-console
+    console.log(`User ${socket.id} connected`);
+  }
+
+  socket.on('disconnect', () => {
+    const user = UserService.findUserById(socket.id);
+
+    if (user && user.currentRoom) {
+      const exitMsg = MessageService.buildMsg(`${user.username} left room`, {
+        id: 0,
+        username: 'Admin',
+        currentRoom: user.currentRoom,
+      });
+
+      socket.leave(user.currentRoom.name);
+      socket.broadcast.to(user.currentRoom.name).emit('message', exitMsg);
+    }
+
+    UserService.removeOne(socket.id);
+    io.emit('users', UserService.users);
```

Comment:
Here, instead of emitting `UserService.users`, you could, in the `removeOne()` from `UserService`, return `users` and use that return here, like:

`const users = UserService.removeOne(socket.id)`
`io.emit('users', users)`

Reply:
It could also be applied in other parts of your code.